File: libs/ui/src/load/loadrequest.ts
import { WireConditionState } from "../bands/wire/conditions/conditions"
import {
	CollectionFieldKey,
	CollectionKey,
	OrderState,
} from "../bands/wire/types"
import { FieldType, SelectListMetadata } from "../wireexports"

type LoadRequest = {
	batchid?: string
	batchnumber?: number
	collection: CollectionKey
	conditions?: WireConditionState[]
	order?: OrderState[]
	name: string
	query?: boolean
	view: string
	batchsize?: number
	requirewriteaccess?: boolean
	fields: LoadRequestField[]
	params?: Record<string, string>
}

type LoadRequestField = {
	id: CollectionFieldKey
	fields?: LoadRequestField[]
	// If this is a View Only field (or on a View-only Wire),
	// then we will have field "metadata" specified inline on the Field
	type?: FieldType
	selectlist?: SelectListMetadata
}

type LoadRequestBatch = {
	wires: LoadRequest[]
	includeMetadata?: boolean
}

export type { LoadRequest, LoadRequestField, LoadRequestBatch }

File: apps/platform/pkg/reqs/loadrequest.go
package reqs

// LoadRequest struct
type LoadRequest struct {
	Collection string                 `json:"collection"`
	Wire       string                 `json:"wire"`
	Type       string                 `json:"type"`
	Fields     []LoadRequestField     `json:"fields"`
	Conditions []LoadRequestCondition `json:"conditions"`
}

// GetCollection function
func (lr *LoadRequest) GetCollection() string {
	return lr.Collection
}

// GetWire function
func (lr *LoadRequest) GetWire() string {
	return lr.Wire
}

// LoadRequestField struct
type LoadRequestField struct {
	ID     string             `json:"id"`
	Fields []LoadRequestField `json:"fields"`
}

// LoadRequestCondition struct
type LoadRequestCondition struct {
	Field       string      `json:"field"`
	Value       interface{} `json:"value"`
	ValueSource string      `json:"valueSource"`
	Type        string      `json:"type"`
	Operator    string      `json:"operator"`
	LookupWire  string      `json:"lookupWire"`
	LookupField string      `json:"lookupField"`
}
